Rust/mechanics_fast/src/lib.rs

```diff
@@ -37,6 +37,22 @@ fn compute_gravitational_acceleration_many_on_one(masses: &[f64], positions: &[V
     acceleration
 }
 
+fn compute_gravitational_acceleration_many_on_many(masses: &[f64], positions: &[Vector3<f64>]) -> Vec<Vector3<f64>> {
+    let mut accelerations = vec![Vector3::new(0.0, 0.0, 0.0); masses.len()];
+
+    for i in 0..masses.len() {
+        for j in 0..masses.len() {
+            if i != j {
+                let displacement = positions[j] - positions[i];
+                let grav_acceleration = compute_gravitational_acceleration_one_on_one(displacement, masses[j]);
+                accelerations[i] += grav_acceleration;
+            }
+        }
```

Comment:
Looks like now the inner `for` loop is basically the same as `compute_gravitational_acceleration_many_on_one`. Could we just call that method instead?
```suggestion
        accelerations[i] = compute_gravitational_acceleration_many_on_one(masses, positions, i);
```

Reply:
can you give me the full function?